instill/model/v1alpha/model.proto

```diff
@@ -4,153 +4,341 @@ package instill.model.v1alpha;
+  // Model description
+  optional string description = 4 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model definition UUID
+  string model_definition = 5 [ (google.api.resource_reference).type = "api.instill.tech/ModelDefinition", (google.api.field_behavior) = IMMUTABLE ];
+  // Model definition configuration
+  Spec configuration = 6 [ (google.api.field_behavior) = IMMUTABLE ];
+  // Model visibility including public or private
+  Visibility visibility = 7 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model full name (i.e., "{user.login}/{model.display_name}")
+  string full_name = 8 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model owner
+  oneof oneof_owner {
+    // Resource name of an organization, e.g., "users/instill".
+    string user = 9 [ (google.api.resource_reference).type = "api.instill.tech/User", (google.api.field_behavior) = OUTPUT_ONLY ];
+    // Resource name of an organization
+    string org = 10 [ (google.api.resource_reference).type = "api.instill.tech/Organization", (google.api.field_behavior) = OUTPUT_ONLY ];
+  };
+  // Model create time
+  google.protobuf.Timestamp create_time = 11 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model update time
+  google.protobuf.Timestamp update_time = 12 [ (google.api.field_behavior) = OUTPUT_ONLY ];
 }
 
-// ReadinessResponse represents a response for a service readiness status
-message ReadinessResponse {
-  // ServingStatus enumerates the status of a queried service
-  enum ServingStatus {
-    // Serving status: UNSPECIFIED
-    SERVING_STATUS_UNSPECIFIED = 0;
-    // Serving status: SERVING
-    SERVING_STATUS_SERVING = 1;
-    // Serving status: NOT SERVING
-    SERVING_STATUS_NOT_SERVING = 2;
+// ModelVersion represents one deployable instance of a model
+message ModelInstance {
+  option (google.api.resource) = {
+    type: "api.instill.tech/ModelInstance",
+    pattern: "models/{model}/instances/{instance}"
+  };
+
+  // Task enumerates the task type of a model instance
+  enum Task {
+    // Task: UNSPECIFIED
+    TASK_UNSPECIFIED = 0;
+    // Task: CLASSIFICATION
+    TASK_CLASSIFICATION = 1;
+    // Task: DETECTION
+    TASK_DETECTION = 2;
+  }
+  // Status enumerates the status of a model instance
+  enum Status {
+    // Status: UNSPECIFIED
+    STATUS_UNSPECIFIED = 0;
+    // Status: OFFLINE
+    STATUS_OFFLINE = 1;
+    // Status: ONLINE
+    STATUS_ONLINE = 2;
+    // Status: ERROR
+    STATUS_ERROR = 3;
   }
 
-  // Status is the value of ServingStatus enum type
-  ServingStatus status = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Resource name. It must have the format of "models/{model}/instances/{instance}"
+  string name = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model instance UUID
+  string id = 2 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model instance display name, this is used to construct the resource name.
+  string display_name = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
```

Comment:
@xiaofei-du this field should be required when creating a ModelIstance right?

Reply:
Check the previous answer. `CreateModelInstance` is removed.